compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -3376,9 +3376,12 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
 
         let (sugg_span, suggestion) = match tcx.sess.source_map().span_to_snippet(args_span) {
             Ok(string) => {
-                let coro_prefix = if string.starts_with("async") {
-                    // `async` is 5 chars long. Not using `.len()` to avoid the cast from `usize`
-                    // to `u32`.
+                let coro_prefix = if string.starts_with("async gen") {
```

Comment:
Also this is still broken b/c it doesn't take into account the fact that the rendered source may have spaces `async     gen {}`